### DISCUSSION TITLE
How to programmatically navigate to a tab

### DISCUSSION THREAD

**Discussion by mnxamdev** [PrismLibrary/Prism#3260] [.NET MAUI]
I used to use a simple tab setup but now I need to use a setup similar to the demo (Tabbed option on root page):
https://github.com/PrismLibrary/Prism/blob/ec09b170e849076ee3dbc0abccb49302ce863404/e2e/Maui/PrismMauiDemo/ViewModels/RootPageViewModel.cs#L23

Similar to the demo example, however, this is my main page: TabA, TabB (with navigation page), TabB, TabC. 
```
prism.CreateWindow(async navigationService =>
                    {
#if ANDROID || IOS
                        await navigationService.CreateBuilder()
                             .AddTabbedSegment(b =>
                                 b.CreateTab(nameof(TabA))
                                     .CreateTab(t => t.AddNavigationPage().AddSegment<TabB>())
                                     .CreateTab(nameof(TabC))
                                     .CreateTab(nameof(TabD))
                             //I get error with this: .SelectedTab(nameof(TabB))
                             )
                             .NavigateAsync(MauiProgram.HandleNavigationError);
#endif
```

How do I programmatically navigate to TabB on start (or anytime really)? If I use .SelectedTab(...) I get "No child page was found with key" error.

Thx!

**Comment by thomasgalliker** [comment]
Did you register the page with name TabB in DI?
- Within UsePrism(p => p.RegisterTypes(c => c....)); you will find have to register pages in the DI container.
- c.RegisterForNavigation< TabB, TabBViewModel>("TabB");
- Make sure all further depenedencies within TabB and TabBViewModel are resolvable

**Comment by mnxamdev** [comment]
@thomasgalliker thanks for chiming in. We're all good there with your recommendations. This isn't a new project and those pages were always registered. We're just changing the navigation from flyout to tab. Once we switched to tab navigation TabB still always worked with SelectedTab(...). When it didn't work was when I did .CreateTab(t => t.AddNavigationPage().AddSegment<TabB>()). 

I did figure this out through inspection though. Unfortunately it doesn't seem the API has the ability to name the navigationpage segment. So through guessing I put in .SelectedTab("NavigationPage") and that fixed it.
